lua/fff/state.lua

```diff
@@ -0,0 +1,200 @@
+---@field matched string Highlight group for matched text
+---@field title string Highlight group for window title
+---@field prompt string Highlight group for prompt text
+---@field active_file string Highlight group for active file line
+---@field frecency string Highlight group for frecency info
+---@field debug string Highlight group for debug messages
+
+---@class ConfigFrecency
+---@field enabled boolean Enable frecency tracking (file access frequency)
+---@field db_path string Path to frecency database file
+
+---@class ConfigLogging
+---@field enabled boolean Enable logging
+---@field log_file string Log file location
+---@field log_level '"debug"'|'"info"'|'"warn"'|'"error"' Logging level
+
+---@class ConfigUI
+---@field wrap_paths boolean Wrap long file paths in the list UI
+---@field wrap_indent integer Indentation spaces for wrapped paths
+---@field max_path_width integer Maximum path width before wrapping
+
+---@class ConfigImagePreview
+---@field enabled boolean Enable image preview (requires terminal support)
+---@field max_width integer Maximum image width in terminal columns
+---@field max_height integer Maximum image height in terminal lines
+
+---@class ConfigIcons
+---@field enabled boolean Enable file icons display
+
+---@class ConfigDebug
+---@field enabled boolean Enable debug mode
+---@field show_scores boolean Show scoring information (toggle with keymap)
+
+---@class FindFileOpts
+---@field git_ignore boolean Whether to respect .gitignore files
+---@field hidden boolean Whether to show hidden files
+---@field git_exclude boolean Whether to respect `.git/info/exclude`
+---@field git_global boolean Whether to respect global gitignore (`core.excludesFile`)
+---@field follow_links boolean Whether to follow symbolic links
+---@field ignore boolean Whether to respect `.ignore` files
+
+---@class Config user configuration
+---@field base_path string Base directory for file indexing
+---@field max_results integer Maximum number of search results to display
+---@field max_threads integer Maximum number of threads for fuzzy search
+---@field prompt string Input prompt symbol
+---@field title string Window title
+---@field ui_enabled boolean Enable UI (default: true)
+---@field width number Window width as fraction of screen
+---@field height number Window height as fraction of screen
+---@field preview ConfigPreview Preview pane configuration
+---@field layout ConfigLayout Layout configuration (alternative to width/height)
+---@field keymaps ConfigKeyMaps Key mappings
+---@field hl ConfigHL Highlight groups
+---@field frecency ConfigFrecency Frecency tracking options
+---@field logging ConfigLogging Logging configuration
+---@field ui ConfigUI UI appearance options
+---@field image_preview ConfigImagePreview Image preview options
+---@field icons ConfigIcons File icon display options
+---@field debug ConfigDebug Debug options
+
+-- State
+---@type State
+M = {
+  initialized = false,
+  last_scan_time = 0,
+  config = {
```

Comment:
can we have the default config isolted somewhere so it is easy to link this from the readme